src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -68,32 +72,56 @@ public ShooterSubsystem() {
-				&& Utils.toRPM(m_flywheel.getSelectedSensorVelocity()) > 0.95 * ShooterConstants.kShooterSpeedRPM) {
+		if ((isShooterPrimed() || m_shootingTimer.get() < 2) && m_blackShooterPID.getSetpoint() > 0
+				&& Utils.toRPM(m_blackFlywheel.getSelectedSensorVelocity()) > 0.95
+						* ShooterConstants.kBlackShooterSpeedRPM) {
 			m_topFeeder.set(ShooterConstants.kTopFeederSpeedFast);
 			if (isShooterPrimed()) {
 				m_shootingTimer.reset();
 			}
+			double greenPidOutput = m_blackShooterPID
+					.calculate(Utils.toRPM(m_greenFlywheel.getSelectedSensorVelocity()));
+			if (m_blackShooterPID.getSetpoint() > 0) {
+				m_greenFlywheel.set(greenPidOutput + m_feedforward.calculate(m_blackShooterPID.getSetpoint()));
+			} else {
+				m_greenFlywheel.set(0);
+			}
+
+			if ((isShooterPrimed() || m_shootingTimer.get() < 2) && m_blackShooterPID.getSetpoint() > 0
+					&& Utils.toRPM(m_greenFlywheel.getSelectedSensorVelocity()) > 0.95
+							* ShooterConstants.kGreenShooterSpeedRPM) {
+				m_topFeeder.set(ShooterConstants.kTopFeederSpeedFast);
+				if (isShooterPrimed()) {
+					m_shootingTimer.reset();
+				}
+			}
 		} else if (!isShooterPrimed() && m_runningIntake) {
 			m_topFeeder.set(ShooterConstants.kTopFeederSpeedSlow);
 		} else if (!m_reversingIntake) {
 			m_topFeeder.set(0);
 		}
 
 		if (OIConstants.kTelemetry) {
-			SmartDashboard.putNumber("Shooter PID Output", pidOutput);
-			SmartDashboard.putNumber("Shooter PID velocity error", m_shooterPID.getVelocityError());
-			SmartDashboard.putNumber("Shooter PID position error", m_shooterPID.getPositionError());
-			SmartDashboard.putNumber("Shooter Feedforward output", m_feedforward.calculate(m_shooterPID.getSetpoint()));
-			SmartDashboard.putNumber("Shooter Power", m_flywheel.get());
-			SmartDashboard.putNumber("Shooter RPM", Utils.toRPM(m_flywheel.getSelectedSensorVelocity()));
+			SmartDashboard.putNumber("Shooter PID Output", blackPidOutput);
+			SmartDashboard.putNumber("Shooter PID velocity error", m_blackShooterPID.getVelocityError());
+			SmartDashboard.putNumber("Shooter PID velocity error", m_greenShooterPID.getVelocityError());
+			SmartDashboard.putNumber("Shooter PID position error", m_blackShooterPID.getPositionError());
+			SmartDashboard.putNumber("Shooter PID position error", m_greenShooterPID.getPositionError());
+			SmartDashboard.putNumber("Shooter Feedforward output",
+					m_feedforward.calculate(m_blackShooterPID.getSetpoint()));
+			SmartDashboard.putNumber("Shooter Feedforward output",
+					m_feedforward.calculate(m_greenShooterPID.getSetpoint()));
+			SmartDashboard.putNumber("Shooter Power", m_blackFlywheel.get());
+			SmartDashboard.putNumber("Shooter Power", m_greenFlywheel.get());
+			SmartDashboard.putNumber("Shooter RPM", Utils.toRPM(m_blackFlywheel.getSelectedSensorVelocity()));
+			SmartDashboard.putNumber("Shooter RPM", Utils.toRPM(m_greenFlywheel.getSelectedSensorVelocity()));
```

Comment:
There are multiple SmartDashboard calls that have the same name when they do different things. This means the one that is called later will always overwrite the one that is called first. (ie lines 113 and 114 both print under "Shooter PID velocity error" when one should be "Shooter PID velocity error Green" and the other should be "Shooter PID velocity error Black"